File: backend/src/main/kotlin/com/sorsix/backend/dto/CaseDto.kt
package com.sorsix.backend.dto

import com.sorsix.backend.domain.cases.Case
import com.sorsix.backend.domain.enums.CaseStatus
import java.time.LocalDateTime

data class CaseDto(
    val id: Long?,
    val bloodType: String?,
    val allergies: String?,
    val description: String?,
    val treatmentPlan: String?,
    val status: CaseStatus,
    val createdAt: LocalDateTime?,
    val updatedAt: LocalDateTime?,
    val patientId: Long,
    val doctorId: Long,
    val examinationsIds: List<Long?>
)

fun Case.toCaseDto() = CaseDto(
    id = this.id,
    bloodType = this.bloodType,
    allergies = this.allergies,
    description = this.description,
    treatmentPlan = this.treatmentPlan,
    status = this.status,
    createdAt = this.createdAt,
    updatedAt = this.updatedAt,
    patientId = this.patient.id,
    doctorId = this.doctor.id,
    examinationsIds = this.examinations.map { it.id }
)

File: frontend/src/app/models/cases/case.dto.ts
import {CaseStatus} from './case-status.enum';

export interface CaseDto {
    id: number;
    public: boolean;
    bloodType?: string | null;
    allergies?: string | null;
    description?: string | null;
    treatmentPlan?: string | null;
    status: CaseStatus;
    createdAt: string; // ISO
    updatedAt: string; // ISO
    patientId: number;
    doctorId: number;
    examinationsIds: number[];
}
